idl2library.cmake

```diff
@@ -0,0 +1,226 @@
+# idl2library(IDLS <files> [BUILD_STATIC])
+# Generate a shared or static library for each IDL in a list.  
+#
+# Example:
+#           idl2library(IDLS idl/example.idl idl/std_doc.idl)
+#           idl2library(IDLS idl/example.idl idl/std_doc.idl BUILD_STATIC)
+#
+
+cmake_minimum_required(VERSION 3.20)
+
+set (IDL_TARGET_SUFFIX "_idl")
+
+#Converts idl file to a target name:  lowervase with no path or extension with IDL_TARGET_SUFFIX appended
+macro(get_idl_target_name idlfile output)
+    get_filename_component(current_idl_shortname ${idlfile} NAME_WE)
+    string(TOLOWER "${current_idl_shortname}" current_idl_shortname)
+    set(temp_output "${current_idl_shortname}")
+    string(APPEND temp_output "${IDL_TARGET_SUFFIX}")
+    set(${output} ${temp_output})
+endmacro()
+
+#Finds all of the input file's (absolute path) dependencies and returns them in IDL_TARGET_DEPENDENCIES
+function(find_idl_dependencies input_file)	
+	unset(search_list)
+	unset(return_list)
+	LIST(APPEND current_list ${input_file}) #Files that need to be searched (full path)
+	LIST(APPEND search_list ${input_file}) #Don't search more than once (full path)
+	
+	while(current_list)
+	    SET(file_list ${current_list})
+		unset(current_list)
+		foreach(current_file ${file_list})
+			set (INCLUDE_REGEX "^[ \t]*\#include[ \t]+\"(.+\\.[I|i][D|d][L|l])\".*$")
```

Comment:
Preprocessor statements can have spaces after the # and includes can use `<>` as well as `""`.  Even then trying to parse anything in an IDL file outside an IDL compiler doesn't seem very robust.

Reply:
Made the two suggested tweaks.  I agree about robustness but it is useful enough to keep and has worked for a while.